app/routes/user_routes.py

```
@@ -53,28 +55,28 @@ def get_user(user_id: int, db: Session = Depends(get_db)):
     if not user:
-        raise HTTPException(status_code=404, detail="User not found")
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="User not found")
     return user
 
 # -----------------------------
 # Login endpoint
 # -----------------------------
-@router.post("/login")
-def login(username: str, password: str, db: Session = Depends(get_db)):
+@router.post("/login", response_model=Response)
+def login(payload: UserLogin, db: Session = Depends(get_db)):
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)